src/Rules/DoctrineKeyValueStyleRule.php

```diff
@@ -111,7 +111,7 @@ public function processNode(Node $callLike, Scope $scope): array
         $tableType = $scope->getType($tableExpr);
         if (! $tableType instanceof ConstantStringType) {
```

Comment:
reading this again 2 things come to mind
1) I think we should work with `$tableType->getConstantStrings()` which would allow things like
```
    public function errorIfTableDoesNotExist(Connection $conn)
    {
        $table = 'ada';
        if (rand(0, 1) === 1) {
           $table = 'ak';
        }
        $conn->assembleNoArrays($table);
    }
```

2) I think we should onyl return a error when debug-mode is enabled. without debug mode we should just accept a not analyzable situation and return without a error

Reply:
Good idea, thanks for the suggestion! I've started working on removing all uses of `instanceof` in this class, but will wait to finalize it until after #563 is resolved (to avoid massive merge conflicts).

Reply:
> 2. I think we should only return a error when debug-mode is enabled. without debug mode we should just accept a not analyzable situation and return without a error

@staabm I'm working on this request now. Would you also like to consider mixed value types as "not analyzable" so that errors such as the following only show up in debug mode?

```
Column 'foo.id' expects value type int, got type mixed
```
Or should I leave these as always enabled? Thanks!